projects/ion/src/lib/pagination/pagination.component.ts

```diff
@@ -136,6 +153,52 @@ export class IonPaginationComponent implements OnChanges, OnInit {
     return (option && option.label) || this.generateLabel(this.itemsPerPage);
   }
 
+  visibleButtons(): Page[] {
```

Comment:
As I talked to you on chat, some improvements could be made in this function, you could separate it on 2 different methods, one to get start page index, and other one to get end page index. Then you could put this validations on variables and use them on ifs.